### DISCUSSION TITLE
Getting original error via exc.orig

### DISCUSSION THREAD

**Discussion by dotX12** [sqlalchemy/sqlalchemy#8023] [Ideas]
When we need to handle an error in depth, just get its details, which constraint caused the error, what table we are working with, we need to know the original error.

In the current realities, this is impossible to do, because. ex.orig is the string that is concatenated from DETAIL + HINT.

This addition partially applies to https://github.com/sqlalchemy/sqlalchemy/discussions/6199.

I used to use this crutch before, but I got tired of doing it, and I needed more data than detail and message (for example, constraint_name)
`error_sql = str(exc.orig).split('\n')[1].split('DETAIL:  ')[1]`

possible solution is attached by me: https://github.com/sqlalchemy/sqlalchemy/pull/8022

**Comment by zzzeek** [comment]
hey there -

in what way does it *not* apply to #6199?  can we please open a bug report and link your PR at #8022 to this?   I'm not the best person to comb through #6199 and gather requirements.    your PR looks like it finds what's being looked for so let's make it so nobody opens this issue again, thanks!

**Reply by dotX12** [reply]
> 

In fact, PR can completely cover #6199.

Your solution: https://github.com/sqlalchemy/sqlalchemy/commit/d01790ebe0b425bf79c4decf24011d0d6234944d
Covers only part of the case, namely, it allows the user to receive sqlstate | pgcode. My PR allows access to the entire error object.

**Reply by dotX12** [reply]
@zzzeek Do I need to create an issue [BUG REPORT] ?
didn't quite understand you.
